crates/rpc/src/dto/receipt.rs

```diff
@@ -372,11 +372,17 @@ impl SerializeForVersion for ExecutionResources<'_> {
 
         let mut serializer = serializer.serialize_struct()?;
 
-        serializer.flatten(&ComputationResources(self.0))?;
-        serializer.serialize_field(
-            "data_availability",
-            &DataAvailability(&self.0.data_availability),
-        )?;
+        if serializer.version < RpcVersion::V08 {
+            serializer.flatten(&ComputationResources(self.0))?;
+            serializer.serialize_field(
+                "data_availability",
+                &DataAvailability(&self.0.data_availability),
+            )?;
+        } else {
+            serializer.serialize_field("l1_gas", &self.0.data_availability.l1_gas)?;
```

Comment:
This should be `&self.0.total_gas_consumed` instead. `data_availability.l1_gas` is just the DA-related gas costs, while here we need the "total" L1 gas as far as I understand.

Reply:
Thanks